tools/doc/allhtml.mjs

```diff
@@ -31,13 +31,35 @@ for (const link of toc.match(/<a.*?>/g)) {
   // Split the doc.
   const match = /(<\/ul>\s*)?<\/\w+>\s*<\w+ id="apicontent">/.exec(data);
 
+  // Get module name
+  const moduleName = href.split('.')[0];
+
   contents += data.slice(0, match.index)
-    .replace(/[\s\S]*?id="toc"[^>]*>\s*<\w+>.*?<\/\w+>\s*(<ul>\s*)?/, '');
+    .replace(/[\s\S]*?id="toc"[^>]*>\s*<\w+>.*?<\/\w+>\s*(<ul>\s*)?/, '')
+    // Prefix TOC links with current module name
+    .replace(/<a href="#([^"]+)"/g, (match, anchor) => {
+      return `<a href="#${moduleName}_${anchor}"`;
```

Comment:
Is this change going to end up breaking any existing external links into the node.js documentation? If so, while I get the intent, I not sure it's something we want to do unless we also include extra anchors with the legacy ids.

Reply:
Originally this PR did, but with 6d3f4608eeb831b860f2456fe57e18e3b38c2f3f the intent is to preserve all existing external links if possible (via [automatic legacy anchor generation](https://github.com/nodejs/node/pull/39304/commits/33929345d07c9d52ff2cf08c9a983eb3a2a330b7#diff-b154a254d3ac7e1cdb7b03817d5c942beb9b59e97fdbbbf394ab7701ac3791e9R424)). Hopefully that will make this PR more acceptable.